packages/generator-react-native-vector-icons/src/app/templates/src/index.ts

```diff
@@ -9,13 +9,17 @@
  * Usage: <<%= className %> name="icon-name" size={20} color="#4F8EF7" />
  */
 
+import type { ComponentProps } from 'react';
+
 import { createIconSet } from '@react-native-vector-icons/<%= commonPackage %>';
 import glyphMap from '../glyphmaps/<%= fontFileName %>.json';
 
-const Icon = createIconSet(glyphMap, {
+export const <%= className %> = createIconSet(glyphMap, {
   postScriptName: '<%= postScriptName %>',
   fontFileName: '<%= fontFileName %>.ttf',
   fontSource: require('../fonts/<%= fontFileName %>.ttf'), // eslint-disable-line @typescript-eslint/no-require-imports, global-require
 });
 
-export default Icon;
+export type <%= className %>IconName = ComponentProps<typeof <%= className %>>['name'];
```

Comment:
```suggestion
export type <%= className %>IconName = keyof typeof glyphMap;
```

this should be faster for TS since it won't have to introspect the component props